pkg/controlplane/xds/manager.go

```diff
@@ -129,9 +132,9 @@ func (m *Manager) DeleteExport(name types.NamespacedName) error {
 func (m *Manager) AddImport(imp *v1alpha1.Import) error {
 	m.logger.Infof("Adding import '%s/%s'.", imp.Namespace, imp.Name)
 
-	if imp.Spec.TargetPort == 0 {
+	if m.crdMode && !meta.IsStatusConditionTrue(imp.Status.Conditions, v1alpha1.ImportTargetPortValid) {
```

Comment:
is there a Status field, other than the conditions, which can be tested (e.g., `allocatedPort`) and if not, should we consider adding one?
I think this stems from the fact that we allow TargetPort in the Spec. Can you please re-explain what's the use case for that? Not sure I understand why a user would care for the random port being used in the data plane, they should just care about the Service (virtual) port, which should always be valid since defined on the new VIP.

Reply:
won't you be able to reconstruct it (indirectly) from the Service corresponding to the Import?